### DISCUSSION TITLE
[WinUI3] How to check, how many pointers are active at the moment?

### DISCUSSION THREAD

**Discussion by adamfierek** [microsoft/WindowsAppSDK#4671] [Q&A]
I need some way to get know, if any pointer is pressed at the moment (and if yes - how many?). Something like Mouse.LeftButton from WPF, which can be called outside of pointer event. How to acheive that?

**Comment by castorix** [comment]
Mouse.LeftButton mainly calls [GetKeyState](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getkeystate) with _VK_LBUTTON_ and WPF handles messages like [WM_POINTERDOWN ](https://learn.microsoft.com/en-us/windows/win32/inputmsg/wm-pointerdown)in the main window procedure (like _SetWindowSubclass_ in WinUI 3) and uses APIs lile [GetPointerInfo ](https://learn.microsoft.com/en-us/windows/win32/api/winuser/nf-winuser-getpointerinfo)
Also [Raw Input](https://learn.microsoft.com/en-us/windows/win32/inputdev/raw-input) could be used

**Comment by codendone** [comment]
There isn't an API available for this in WinAppSDK. Note that `GetPointerInfo` should not be used in WinAppSDK, as handling of input in WinAppSDK is somewhat async from Windows. The pointerId values used by WinAppSDK may also not be the same values used by Windows.

You could track PointerEnter, PointerDown, PointerUp, and PointerLeave events (probably via `AddHandler()` with `handledEventsToo=true` on the root XAML element), if needed.

Would you like to share more about your scenario, in case there is a different feature we should consider adding?

**Reply by adamfierek** [reply]
Yes, I want to detect if after PointerPressed fires, there is some other pointers active (I'm using multitouch screen). I tried to use Pointer events to determinate state of all pointer separately and check states every new PointerPressed event, but it wasn't reliable - sometimes without any reason one of pointers fired Released or Canceled or Exited, so physically I had 4 fingers touching screen but in app i was able to detect only 3 or 2.
